scripts/events.py

```diff
@@ -217,10 +217,9 @@ def one_moon(self):
         self.check_and_promote_leader()
         self.check_and_promote_deputy()
 
-        # Resort if needed
-        if game.ranks_changed_timeskip and game.sort_type == "rank":
-            game.ranks_changed_timeskip = False
-            Cat.sort_cats()
+
+        #Resort
+        Cat.sort_cats()
```

Comment:
Wouldn't this result in a lag if we have many cats? It might only result in unnecessary workload, is it really necessary to resort the cat every moon?

Reply:
Before, the cats were resorted any time there was a rank change and "rank" type sorting was selected. The tag that kept track of  rank changes was not always catching everything. In addition, if a cat died, they would not be placed in starclan properly, since Starclan used cat.dead_for rather than cat.moons for age sorting. 
 Tagging wasn't very effective in reducing lag. As clans grow larger, it's very common to have rank changes every moon, which resulted in re-sorting every moon. If we were to also resort every time a cat died, we would almost certainly be resorting every moon.  The only clans which won't trigger resort every moon would be small. 
 
 The only type of sort which is not going to commonly change every moon is id sorting. I have changed these lines to:
 
 if game.sort_type != "id":
      Cat.sort_cats()

Reply:
Good point.